RHIDP-13141: Custom review pages in Orchestrator (#2136)
redhat-developer/red-hat-developers-documentation-rhdh: modules/extend_orchestrator-in-rhdh/con-workflow-review-pages-for-your-approval-requirements.adoc

:_mod-docs-content-type: CONCEPT

[id="workflow-review-pages-for-your-approval-requirements_{context}"]
= Workflow review pages for your approval requirements

[role="_abstract"]

> You can replace the default Orchestrator review page with your own component to match organizational standards, show warnings or require acknowledgment before run, or integrate with design systems in {product}.

Reviewer: ```suggestion
The {product} Orchestrator displays a review page before executing workflows to verify form data before a workflow executes. You can replace the default Orchestrator review page with your own component to meet organizational standards, show warnings, require acknowledgment before run, or integrate with design systems in {product}.
```

The Orchestrator plugin displays a review page before executing workflows, showing form data for verification. While the default review page works for most use cases, some organizations need custom layouts, in-page warnings or approval prompts, or integration with existing UI components.

> Custom review pages are optional. If you do not implement a custom review component, the Orchestrator continues to use the default review page without any impact on functionality.

Reviewer: ```suggestion
Custom review pages are optional. If you do not implement a custom review component, the Orchestrator continues to use the default review page without any impact on functionality.
```
(Removed the first paragraph as it is redundant with the short description)

> Use a custom review page when you need to:

Reviewer: ```suggestion
Use a custom review page when you need to perform the following actions:
```
( I just thought it was a little ambigious when we don't complete the sentence. I know this is mentioned somewhere in our style guide but I'm not able to find it at the moment)

* Display workflow data in a specific layout that matches your organization's documentation or approval standards
* Show warnings or apply client-side checks before calling the execute callback (or before the workflow runs).
* Include additional context, help text, or documentation for reviewers

> * Integrate with custom UI component libraries or design systems

Reviewer: ```suggestion
* Display workflow data in a specific layout that matches your organization's documentation or approval standards.
* Apply client-side checks or show warnings before the workflow runs.
* Include additional context, such as help text or links to documentation, for reviewers.
* Integrate with custom UI component libraries or design systems.
```

> Custom review pages maintain full compatibility with existing workflows. The same workflow definitions, schemas, and data structures work with both default and custom review pages. You can switch between default and custom review pages without modifying your workflow configurations.

Reviewer: ```suggestion
Custom review pages are compatible with existing workflows. The same workflow definitions, schemas, and data structures work with both default and custom review pages. You can switch between review page types without modifying your workflow configurations.
```

:_mod-docs-content-type: CONCEPT

[id="workflow-review-pages-for-your-approval-requirements_{context}"]
= Workflow review pages for your approval requirements

[role="_abstract"]

> You can replace the default Orchestrator review page with your own component to match organizational standards, add custom validation, or integrate with design systems in {product}.

Reviewer 2: Please replace “add custom validation” with “show warnings or require acknowledgment before run” so we don’t imply a dedicated validation API—custom review only supplies layout/UX

Something like,

You can replace the default Orchestrator review page with your own component to match organizational standards, show warnings or require acknowledgment before run, or integrate with design systems in {product}.

> The Orchestrator plugin displays a review page before executing workflows, showing form data for verification. While the default review page works for most use cases, some organizations need custom layouts, additional validation messages, or integration with existing UI components.

Reviewer 2: Swap “additional validation messages” for “in-page warnings or approval prompts” so the doc describes UI you implement in the custom component, not a separate Orchestrator validation API.

The Orchestrator plugin displays a review page before executing workflows, showing form data for verification. While the default review page works for most use cases, some organizations need custom layouts, in-page warnings or approval prompts, or integration with existing UI components.

Custom review pages are optional. If you do not implement a custom review component, the Orchestrator continues to use the default review page without any impact on functionality.

Use a custom review page when you need to:

* Display workflow data in a specific layout that matches your organization's documentation or approval standards

> * Add validation warnings or business rule checks before workflow execution

Reviewer 2: ```suggestion
* Show warnings or apply client-side checks before calling execute (or before workflow execution).
```

* Include additional context, help text, or documentation for reviewers
* Integrate with custom UI component libraries or design systems

> * Provide role-based review interfaces with different information for different user types

Reviewer 2: Not supported today: Role-based or identity-specific review content from Orchestrator—the review component receives only form schema, data, and callbacks, not user or role. We can remove this line.
